src/snakemake/ioutils/branch.py

```
@@ -46,13 +46,17 @@ def do_branch_cases(wildcards):
         try:
             selected_case = cases[res]
         except KeyError:
+            if otherwise is not None:
+                return handle_callable(otherwise, wildcards)
             raise KeyError(f"Key {res} not found in given cases of branch function")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)